packages/medusa/src/api/routes/admin/products/add-option.ts

```diff
@@ -42,13 +42,16 @@ export default async (req, res) => {
   )
 
   const productService: ProductService = req.scope.resolve("productService")
+  const pricingService: PricingService = req.scope.resolve("pricingService")
 
   await productService.addOption(id, validated.title)
-  const product = await productService.retrieve(id, {
+  const rawProduct = await productService.retrieve(id, {
     select: defaultAdminProductFields,
     relations: defaultAdminProductRelations,
   })
 
+  const [product] = await pricingService.setAdditionalPrices([rawProduct], {})
```

Comment:
Previously the `setAdditionalPrices` logic was handled within the ProductService. I am suggesting moving it to avoid coupling products and pricing too tightly.

Also, the pricing service depends on the ProductService so injecting it into the ProductService would create a circular dependency.

Reply:
From what I can see, the `productService` is not used in the `pricingService` so that dependency could be removed.
Also, Some method are taking an id, such as `getProductVariantPricing` which should probably take a variant instead of just the id, same goes for `getProductPricing` which should take a product including the variants instead of the product id, that way the variant query will then goes back to the consumer that knows what to query.
in the end, the pricing service would not have any dependency to either productService or productVariantService and therefore not circular dependency as well :)

Reply:
That is the kind of architectural subject that are better to discuss on a phone call 😅 for now you can choose the approach you prefer 😊